src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -312,11 +324,11 @@ private static bool IsIReferenceArray(Type implementationType)
             // PropertySet and ValueSet can return IReference<String> but Nullable<String> is illegal
             if (runtimeClassName == "Windows.Foundation.IReference`1<String>")
             {
-                return (IInspectable obj) => new ABI.System.Nullable<String>(obj.ObjRef);
+                return CreateReferenceCachingFactory((IInspectable obj) => new ABI.System.Nullable<String>(obj.ObjRef));
```

Comment:
Technically I think ABI.System.Nullable would have handled keeping the objref alive I think, but this doesn't hurt either.